ampd/src/stellar/rpc_client.rs

```diff
@@ -51,19 +51,26 @@ impl TryFrom<(Hash, GetTransactionResponse)> for TxResponse {
                 contract_events: vec![],
             });
         }
+
         let contract_events = match response.result_meta.as_ref() {
             Some(TransactionMeta::V4(_)) => {
                 let contract_events = response.events.contract_events;
                 let operation_count = contract_events.len();
 
+                // Protocol V23: Soroban contract calls should have exactly one operation.
+                // This prevents potential security issues from processing multiple operations
+                // or malformed transactions that could lead to event ordering confusion.
                 if operation_count != EXPECTED_SOROBAN_OPERATION_COUNT {
                     return Err(TxParseError::InvalidOperationCount {
                         expected: EXPECTED_SOROBAN_OPERATION_COUNT,
                         actual: operation_count,
                     });
                 }
 
-                contract_events.into_iter().next().unwrap_or_default()
+                contract_events
+                    .first()
+                    .expect("soroban operation not found")
+                    .clone()
```

Comment:
there are a bunch of lint warnings below, address those? any reason the gh actions aren't catching them, maybe some config needs to be tuned?